packages/components/src/popover/index.js

```diff
@@ -181,9 +182,61 @@ const Popover = (
 		};
 	}, [ ownerDocument ] );
 
+	const avoidOverflow = useMemo(
+		() =>
+			__unstableAvoidOverflow
+				? {
+						name: 'avoidOverflow',
+						async fn( middlewareProps ) {
+							const {
+								x,
+								y,
+								rects,
+								placement: currentPlacement,
+							} = middlewareProps;
+							const isTopPlacement =
+								currentPlacement.includes( 'top' );
```

Comment:
This in general lives in Popover but seems too specific for our block toolbar issue. Ideally  we would  consolidate the other flags to better adjust the core middlewares or our middleware would be probably in the end to handle any kind of overflow.. Not there yet to suggest a solution though..